hrm-domain/hrm-core/profile/profileService.ts

```diff
@@ -39,11 +39,12 @@ import { ITask } from 'pg-promise';
 export {
   Identifier,
   Profile,
-  getIdentifierWithProfiles,
   ProfileListConfiguration,
   SearchParameters,
 } from './profileDataAccess';
 
+const sanitizeIdentifier = (i: string) => i.replace(' ', '');
```

Comment:
I wish we had nominal typing in TS :sweat_smile:. I've been thinking how we might enforce this kind of rules, and some alternatives are:
- Use runtime checkers (probably the best idea here).
- Force the data access layer to accept "refined types", something like `type ValidIdentifier = { identifier: string; __valid: true }`, so the services need to always sanitize the strings before passing them down.

Not sure if any is worth it but opening the door for that discussion.

Reply:
Yeah, I agree. Your first suggested strategy is applicable, not only here but, EVERYWHERE we depend on external data.

Reply:
As I said in my other comment, I think for this particular case, longer term 'sanitization' needs to be done in clients that are more aware of the particular requirements of the channel the identifier comes from - to HRM, identifiers should probably just be identifiers